#
# This patch adds a toggle option to the xfwm4-tweaks dialog that
# forces the resize hint box to display on all resize operations,
# not just on windows with fixed resize increment hints.
#
# Note:
#   Unfortunately, the Glade UI file is precompiled in the distribution
#   tarball, so it will need to be regenerated somehow.
#
--- settings-dialogs/tweaks-settings.c
+++ settings-dialogs/tweaks-settings.c
@@ -196,6 +196,7 @@
     GtkWidget *placement_ratio_scale = GTK_WIDGET (gtk_builder_get_object (builder, "placement_ratio_scale"));
     GtkWidget *placement_center_option = GTK_WIDGET (gtk_builder_get_object (builder, "placement_center_option"));
     GtkWidget *placement_mouse_option = GTK_WIDGET (gtk_builder_get_object (builder, "placement_mouse_option"));
+    GtkWidget *show_size_on_resize_check = GTK_WIDGET (gtk_builder_get_object (builder, "show_size_on_resize_check"));
 
     /* Compositing tab */
     GtkWidget *use_compositing_check = GTK_WIDGET (gtk_builder_get_object (builder, "use_compositing_check"));
@@ -383,6 +384,10 @@
                             "/general/placement_ratio",
                             G_TYPE_INT,
                             (GObject *) gtk_range_get_adjustment (GTK_RANGE (placement_ratio_scale)), "value");
+    xfconf_g_property_bind (xfwm4_channel,
+                            "/general/show_size_on_resize",
+                            G_TYPE_BOOLEAN,
+                            (GObject *)show_size_on_resize_check, "active");
 
     /* Compositing tab */
     xfconf_g_property_bind (xfwm4_channel,
--- settings-dialogs/xfwm4-tweaks-dialog.glade
+++ settings-dialogs/xfwm4-tweaks-dialog.glade
@@ -760,6 +760,21 @@
                     <property name="position">2</property>
                   </packing>
                 </child>
+                <child>
+                  <object class="GtkCheckButton" id="show_size_on_resize_check">
+                    <property name="label" translatable="yes">Show window size when resizing</property>
+                    <property name="visible">True</property>
+                    <property name="can_focus">True</property>
+                    <property name="receives_default">False</property>
+                    <property name="use_underline">True</property>
+                    <property name="draw_indicator">True</property>
+                  </object>
+                  <packing>
+                    <property name="expand">False</property>
+                    <property name="fill">False</property>
+                    <property name="position">3</property>
+                  </packing>
+                </child>
               </object>
               <packing>
                 <property name="position">4</property>
--- src/moveresize.c
+++ src/moveresize.c
@@ -1710,7 +1710,9 @@
 
     passdata.poswin = NULL;
 #ifndef SHOW_POSITION
-    if ((c->size->width_inc > 1) || (c->size->height_inc > 1))
+    /* Show the window size if the window has a size increment hint. */
+    if (c->screen_info->params->show_size_on_resize ||
+        ((c->size->width_inc > 1) || (c->size->height_inc > 1)))
 #endif /* SHOW_POSITION */
     {
         passdata.poswin = poswinCreate(screen_info->gscr);
--- src/settings.c
+++ src/settings.c
@@ -716,6 +716,7 @@
         {"show_dock_shadow", NULL, G_TYPE_BOOLEAN, TRUE},
         {"show_frame_shadow", NULL, G_TYPE_BOOLEAN, TRUE},
         {"show_popup_shadow", NULL, G_TYPE_BOOLEAN, TRUE},
+        {"show_size_on_resize", NULL, G_TYPE_BOOLEAN, TRUE},
         {"snap_resist", NULL, G_TYPE_BOOLEAN, TRUE},
         {"snap_to_border", NULL, G_TYPE_BOOLEAN, TRUE},
         {"snap_to_windows", NULL, G_TYPE_BOOLEAN, TRUE},
@@ -815,6 +816,8 @@
         getBoolValue ("show_frame_shadow", rc);
     screen_info->params->show_popup_shadow =
         getBoolValue ("show_popup_shadow", rc);
+    screen_info->params->show_size_on_resize =
+        getBoolValue ("show_size_on_resize", rc);
     screen_info->params->snap_to_border =
         getBoolValue ("snap_to_border", rc);
     screen_info->params->snap_to_windows =
@@ -1321,6 +1324,10 @@
                     screen_info->params->show_popup_shadow = g_value_get_boolean (value);
                     reloadScreenSettings (screen_info, UPDATE_FRAME);
                 }
+                else if (!strcmp (name, "show_size_on_resize"))
+                {
+                    screen_info->params->show_size_on_resize = g_value_get_boolean (value);
+                }
                 else if (!strcmp (name, "snap_resist"))
                 {
                     screen_info->params->snap_resist = g_value_get_boolean (value);
--- src/settings.h
+++ src/settings.h
@@ -220,6 +220,7 @@
     gboolean show_dock_shadow;
     gboolean show_frame_shadow;
     gboolean show_popup_shadow;
+    gboolean show_size_on_resize;
     gboolean snap_resist;
     gboolean snap_to_border;
     gboolean snap_to_windows;
